zebra-chain/src/transaction.rs

```diff
@@ -97,6 +97,8 @@ pub enum Transaction {
     },
     /// A `version = 5` transaction, which supports `Sapling` and `Orchard`.
     V5 {
+        /// The Consensus branch ID for this transaction.
+        consensus_branch_id: ConsensusBranchId,
```

Comment:
Should we use a NetworkUpgrade instead?

Benefits:
- most bad branch IDs are detected during deserialisation
- interfaces to sighash v5 would be simpler, because it can just take a network upgrade
- we can't represent a wide range of branch IDs (and we don't want to)
- some of the extra code we just added would disappear

Drawbacks:
- it's slightly different from the underlying data